spec/cyprus_postal_codes/client/search_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+RSpec.describe CyprusPostalCodes::Client::Search do
+  describe "search" do
+    subject(:client) do
+      Class.new do
+        include CyprusPostalCodes::Client::Search
+
+        def get(resource, options); end
+      end
+    end
+
+    before { stub_const("SearchClient", client) }
+
+    let(:service) { SearchClient.new }
+
+    it "calls the `search` endpoint" do
+      allow(service).to receive(:get)
+
+      service.search(district: "lefkosia", param: "arch")
+
+      expect(service).to have_received(:get).with("search", district: "lefkosia", param: "arch", area: nil,
+                                                            page_token: nil)
+    end
+
+    it "expects the :district, :param to be set" do
```

Comment:
Maybe update all examples to follow a pattern like below for consistency, as we don't use colon everywhere anyway

```suggestion
    it "expects `district` and `param` to be set" do
```